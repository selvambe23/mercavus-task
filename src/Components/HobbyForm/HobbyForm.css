.userHobbyAddform{
    display: flex;
    align-items: flex-start;
    flex-wrap: nowrap;
    flex-direction: column;
}
.userHobbyAddform button {
    margin-bottom: 0.5em;
    background: #fafafa;
    color: #333;
    font-size: 1em;
    font-weight: bold;
    padding: 14px 25px;
    border-radius: 10px;
    text-decoration: none;
    display: inline-block;
  }
select {
    display: inline-block;
    height: 55px;
    font-size: 15px;
    margin-right: 10px;
}
.nameBox {
    padding: 15px 15px;
    border: 2px solid #aaa;
    border-radius: 10px;
    margin-right: 20px;
    font-size: 18px;
  }
  .passionBox {
    width: 100%;
    display: flex;
    flex-direction: column;
    margin-right: 10px;
    margin-bottom: 15px;
}
.hobbyItems{
    display: flex;
    align-items: flex-start;
    flex-wrap: nowrap;
    flex-direction: row;
}
.hobbyItems div {
    width: 50%;
}