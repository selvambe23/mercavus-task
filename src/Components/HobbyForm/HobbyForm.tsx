import React, { useState } from "react";
import "./HobbyForm.css";

const UserForm: React.FC<{ addHobbiesFunc: any }> = ({ addHobbiesFunc }) => {
  const [hobby, setHobby] = useState<any>("");
  const [year, setYear] = useState<any>("");
  const [passion, setPassionLevel] = useState<any>("default");

  const UserForm = (
    <div className="userHobbyAddform">
      <div className="passionBox">
        <select value={passion} onChange={e => setPassionLevel(e.target.value)}>
          <option key="0" value="default">
            Select Passion Level
          </option>
          <option key="1" value="Medium">
            Medium
          </option>
          <option key="2" value="Low">
            Low
          </option>
          <option key="3" value="High">
            High
          </option>
          <option key="4" value="Very-High">
            Very-High
          </option>
        </select>
      </div>
      <div className="passionBox">
        <input
          type="text"
          placeholder="Enter User Hobby"
          onChange={e => setHobby(e.target.value)}
          value={hobby}
          className="nameBox"
        />
      </div>
      <div className="passionBox">
        <input
          type="text"
          placeholder="Enter Year"
          value={year}
          onChange={e => setYear(e.target.value)}
          className="nameBox"
        />
      </div>
      <button onClick={e => addHobbiesFunc(e, hobby, year, passion)}>
        Add
      </button>
    </div>
  );
  return UserForm;
};

export default UserForm;
