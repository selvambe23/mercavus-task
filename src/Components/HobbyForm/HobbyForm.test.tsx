import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from "react-test-renderer";
import HobbyForm from './HobbyForm';

describe('HobbyForm Component', () => {
  it('renders snapshots, too', () => {
    const addHobbiesFunc = jest.fn();
    const wrapper = shallow(<HobbyForm addHobbiesFunc={addHobbiesFunc} />);
    expect(wrapper.find('.userHobbyAddform')).toBeDefined();

    const tree = renderer
      .create(<HobbyForm addHobbiesFunc={addHobbiesFunc} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
});