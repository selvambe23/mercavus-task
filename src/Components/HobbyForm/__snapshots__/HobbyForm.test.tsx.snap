// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HobbyForm Component renders snapshots, too 1`] = `
<div
  className="userHobbyAddform"
>
  <div
    className="passionBox"
  >
    <select
      onChange={[Function]}
      value="default"
    >
      <option
        value="default"
      >
        Select Passion Level
      </option>
      <option
        value="Medium"
      >
        Medium
      </option>
      <option
        value="Low"
      >
        Low
      </option>
      <option
        value="High"
      >
        High
      </option>
      <option
        value="Very-High"
      >
        Very-High
      </option>
    </select>
  </div>
  <div
    className="passionBox"
  >
    <input
      className="nameBox"
      onChange={[Function]}
      placeholder="Enter User Hobby"
      type="text"
      value=""
    />
  </div>
  <div
    className="passionBox"
  >
    <input
      className="nameBox"
      onChange={[Function]}
      placeholder="Enter Year"
      type="text"
      value=""
    />
  </div>
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
